src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -197,6 +212,18 @@ var DOMRenderer = ReactFiberReconciler({
     return true;
   },
 
+  commitInitialEffects(
+    domElement : Instance,
+    type : string,
+    newProps : Props,
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
+  ) : void {
+    if (shouldAutoFocusHostComponent(type, newProps)) {
+      (domElement : any).focus();
+    }
```

Comment:
I attempted to explain my reasoning in my previous comment. Some of it is subjective.

* That change would introduce asymmetry between `prepareUpdate` and `commitUpdate`. `commitUpdate` would be called for newly-mounted components as well as pre-existing ones but `prepareUpdate` would only be called for existing ones. (Technically we _could_ call `prepareUpdate` during initial mount as well but it seems unnecessary.)
* `commitUpdate` would be called during different times, depending on whether the component was initially mounting (in which case it would be called during `commitAllLifeCycles`) or whether it was an updated to a pre-existing component (in which case it would be called during `commitWork`).
* Adding checks to handle possible null values for `oldProps` for all renderers just to support a specific on-mount use case for one renderer felt (subjectively) wrong.